### PR TITLE
Use `libreload-packit` to handle live reload

### PR DESCRIPTION
**Pull request by joshuatcasey**
Use `libreload-packit` to handle live reload.

Blocked on #269 which is why this diff is so large.

### CODE REVIEW

**Comment by joshuatcasey** [comment]
@paketo-buildpacks/nodejs-maintainers Can you review please?

Since this PR modifies `go.mod` / `go.sum` it's hard to keep it  cleanly ahead of `main`.

**Review by TisVictress — COMMENTED** [review]
LGTM! I'll approve once updated

**Review by TisVictress — APPROVED** [review]

**Pull request merged**
